### DISCUSSION TITLE
Events API can be polled, but how about a webhook?

### DISCUSSION THREAD

**Discussion by RickCogley** [outline/outline#1789] [Ideas]
The events API can be polled periodically, to get a change log or audit log which orgs might want for various reasons, but I wonder if you could consider adding a webhook target to push all events to, so a log could be collected real time.

**Comment by tommoor** [comment]
Hey Rick, would love to offer webhooks – there was an issue for it for a while but it just has never really been requested (up until now!) so it got closed out as not being important enough especially as polling is a reasonable fallback for most things.

**Reply by RickCogley** [reply]
Thanks & got it, @tommoor. Yep, polling works just fine. One easy way to do it is to use a service like <https://webhook.site>, which allows scheduling, to pull every so often then turn around and push to whatever db or file you want to log to. That way you don't even need to bother with a server & cron.

**Reply by almereyda** [reply]
Webhooks are also very useful for developing bots, that integrate Outline with other ecosystems, like Matrix #1486 or Taiga #1911

For documentation purposes, the old issue was #423.

**Reply by rossmeissl** [reply]
@tommoor we're very interested in webhooks. If you want help, my team and I could pitch in on a PR. But probably would need some general direction from you on how you want it done.
